### ISSUE TITLE
After opening a few tabs in background and closing them, the app will crash

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#10467]
## Steps to reproduce
1. Launch Fennec
2. Long press on a shortcut and open in new tab
3. Long press on a shortcut and open in new tab
4. Background Fennec ~ 10 seconds
5. Restore Fennec
6. Open tabs tray
7. Close the last tab
### Expected behavior
- The tabs should close.
### Actual behavior
- Fennec crashes. After re-opening the app, on some occasions, repeating steps 6) and 7) will cause the crash to repeat.
### Device & build information
* Device: iPhone 12 Pro (15.4)
* Build version:  Fennec f1f25eee956d468f4b0e37537a5391cf46419486
### Notes
Attachments: 

https://user-images.githubusercontent.com/32130829/162973819-6f69a7d4-3618-402b-858f-22c3b3531122.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4088)

**Comment by SimonBasca** [comment]
Might be a duplicate of https://github.com/mozilla-mobile/firefox-ios/issues/10447

**Comment by SimonBasca** [comment]
Log from Xcode:

```
2022-04-12 16:22:43.738710+0300 Client[74572:4023543] [CollectionView] Layout attributes <UICollectionViewLayoutAttributes: 0x1240f0cd0> index path: (<NSIndexPath: 0x97d03a1a5b4a0566> {length = 2, path = 0 - 1}); frame = (0 120; 362.667 112);  were received from the layout <UICollectionViewCompositionalLayout: 0x125ac35f0> but are not valid for the data source counts. Attributes will be ignored.
2022-04-12 16:22:43.738871+0300 Client[74572:4023543] *** Assertion failure in -[UICollectionView _updateVisibleCellsNow:], UICollectionView.m:4755
2022-04-12 16:22:43.742258+0300 Client[74572:4023543] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'UICollectionView internal inconsistency: missing final attributes for cell <Client.FxHomeHorizontalCell: 0x125a4dbb0; baseClass = UICollectionViewCell; frame = (0 120; 362.667 112); tag = 1; layer = <CALayer: 0x2801167c0>>; initial attributes: <UICollectionViewLayoutAttributes: 0x125a5db60> index path: (<NSIndexPath: 0x97d03a1a5b4a0566> {length = 2, path = 0 - 1}); frame = (0 120; 362.667 112); ; layout query: <UICollectionViewLayoutAttributes: 0x1240f0cd0> index path: (<NSIndexPath: 0x97d03a1a5b4a0566> {length = 2, path = 0 - 1}); frame = (0 120; 362.667 112); ; collection view: <UICollectionView: 0x10c8d8000; frame = (0 0; 390 136); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x280ebf120>; animations = { bounds.origin=<CABasicAnimation: 0x28004a860>; bounds.size=<CABasicAnimation: 0x28004a7e0>; position=<CABasicAnimation: 0x28004a080>; }; layer = <CALayer: 0x2801a35c0>; contentOffset: {-16, -8}; contentSize: {390, 232}; adjustedContentInset: {8, 16, 8, 16}; layout: <UICollectionViewCompositionalLayout: 0x125ac35f0>; dataSource: <Client.FxHomeJumpBackInCollectionCell: 0x125a6a060; baseClass = UICollectionViewCell; frame = (0 543; 390 136); animations = { position=<CABasicAnimation: 0x28004aec0>; bounds.origin=<CABasicAnimation: 0x280048940>; bounds.size=<CABasicAnimation: 0x280049a40>; }; layer = <CALayer: 0x2801f44c0>>>'
*** First throw call stack:
(0x1a8645d78 0x1c12aa734 0x1a9ecb1f0 0x1aac41fd8 0x1aab96314 0x1aaba8844 0x1ac31e09c 0x1aac0c5f4 0x1aaba1748 0x1aaec703c 0x1ab08bfd4 0x1aaba1748 0x1aac83ba8 0x1aae067a8 0x1ab05c94c 0x1aaea30b4 0x100780194 0x100355ff8 0x1020f0c6c 0x1020f27bc 0x102102c68 0x1021027a4 0x1a85fe2f0 0x1a85b81f4 0x1a85cb6b8 0x1c4665374 0x1aaf30e88 0x1aacb25ec 0x10091b88c 0x101d05ce4)
libc++abi: terminating with uncaught exception of type NSException
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'UICollectionView internal inconsistency: missing final attributes for cell <Client.FxHomeHorizontalCell: 0x125a4dbb0; baseClass = UICollectionViewCell; frame = (0 120; 362.667 112); tag = 1; layer = <CALayer: 0x2801167c0>>; initial attributes: <UICollectionViewLayoutAttributes: 0x125a5db60> index path: (<NSIndexPath: 0x97d03a1a5b4a0566> {length = 2, path = 0 - 1}); frame = (0 120; 362.667 112); ; layout query: <UICollectionViewLayoutAttributes: 0x1240f0cd0> index path: (<NSIndexPath: 0x97d03a1a5b4a0566> {length = 2, path = 0 - 1}); frame = (0 120; 362.667 112); ; collection view: <UICollectionView: 0x10c8d8000; frame = (0 0; 390 136); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x280ebf120>; animations = { bounds.origin=<CABasicAnimation: 0x28004a860>; bounds.size=<CABasicAnimation: 0x28004a7e0>; position=<CABasicAnimation: 0x28004a080>; }; layer = <CALayer: 0x2801a35c0>; contentOffset: {-16, -8}; contentSize: {390, 232}; adjustedContentInset: {8, 16, 8, 16}; layout: <UICollectionViewCompositionalLayout: 0x125ac35f0>; dataSource: <Client.FxHomeJumpBackInCollectionCell: 0x125a6a060; baseClass = UICollectionViewCell; frame = (0 543; 390 136); animations = { position=<CABasicAnimation: 0x28004aec0>; bounds.origin=<CABasicAnimation: 0x280048940>; bounds.size=<CABasicAnimation: 0x280049a40>; }; layer = <CALayer: 0x2801f44c0>>>'
terminating with uncaught exception of type NSException
```

**Comment by lmarceau** [comment]
This should be fixed by https://github.com/mozilla-mobile/firefox-ios/pull/11605

**Comment by data-sync-user** [comment]
➤ Simion Basca commented:

Marking as verified fix on main 0edd33a647a5efff980a9c4aa865d5c0d936a54e
